asabiris/formatter/jinja/service.py

```diff
@@ -1,39 +1,75 @@
+    """
 
+    def __init__(self, app, service_name="JinjaService"):
+        """
+        Initialize the JinjaFormatterService.
 
-class JinjaFormatterService(asab.Service, FormatterABC):
+        :param app: The application instance.
+        :param service_name: The name of the service.
+        """
+        super().__init__(app, service_name)
+
+        # Load variables from the configuration section named 'variables'
+        try:
+            self.Variables = {option.upper(): asab.Config.get('variables', option) for option in asab.Config.options('variables')}
+        except configparser.NoSectionError:
+            self.Variables = {}
+
+        # Load variables from JSON file if specified in the configuration
+        json_path = asab.Config.get('jinja', 'variables', fallback=None)
+        if json_path and os.path.exists(json_path):
+            with open(json_path, 'r') as json_file:
+                self.Variables.update({k.upper(): v for k, v in json.load(json_file).items()})
+
+    def __getitem__(self, key):
+        """
+        Retrieve the value of a variable using the square bracket notation.
+
+        :param key: The name of the variable.
+        :return: The value of the variable or None if not found.
+        """
+        return self.Variables.get(key.upper(), None)
+
+    async def format(self, template_path, template_params):
+        """
+        Format a given template using Jinja2.
+
+        :param template_path: The path to the Jinja2 template.
+        :param template_params: The parameters to be used in the template.
+        :return: The formatted template.
+        """
+        # Combine the provided template_params with the loaded Variables
+        jinja_variables = collections.ChainMap(template_params, self.Variables)
 
-	def __init__(self, app, service_name="JinjaService"):
-		super().__init__(app, service_name)
-		try:
-			self.Variables = {option: asab.Config.get('variables', option) for option in asab.Config.options('variables')}
-		except configparser.NoSectionError:
-			self.Variables = {}
+        # Read the template from the specified template_path
+        template_io = await self.App.LibraryService.read(template_path)
+        if template_io is None:
+            raise PathError("Template '{}' not found".format(template_path))
 
-	async def format(self, template_path, template_params):
-		jinja_variables = collections.ChainMap(template_params, self.Variables)
-		template_io = await self.App.LibraryService.read(template_path)
-		if template_io is None:
-			raise PathError("Template '{}' not found".format(template_path))
+        # Create a Jinja2 template from the read content
+        template = jinja2.Template(template_io.read().decode('utf-8'))
 
-		template = jinja2.Template(template_io.read().decode('utf-8'))
+        # Convert keys with dot notation into nested dictionaries
+        template_params = utils.create_nested_dict_from_dots_in_keys(jinja_variables)
```

Comment:
Is it tested?

I see that you prepared that `__getitem__` method.
But you are not applying that anywhere.

How do you apply that on `template_params`?

I guess you need to create some kind of wrapper class.